src/stages/stage3_events.c

```diff
@@ -741,939 +81,689 @@ Boss* stage3_spawn_scuttle(cmplx pos) {
+							.pos = e->pos + dir * 10,
+							.color = RGB(1.0, 0.6, 0.3),
+							.move = move_linear(dir * (1.0 + 0.5 * sin(a))),
+							);
+				}
 
-	if(laser) {
-		p->args[3] = laser->prule(laser, time - p->args[1]) - p->pos;
-	}
+				play_sound("shot1");
+				WAIT(shot_interval);
+			}
 
-	p->angle = carg(p->args[3]);
-	p->pos = p->pos + p->args[3];
+		case 2:
+			e->move.attraction = 0.03;
+			WAIT(60);
 
-	return ACTION_NONE;
-}
+			for (int i = 0; i < 80; ++i) {
+				double a = global.timer/sqrt(global.diff);
+				cmplx dir = cdir(a);
 
-static void wriggle_ignite_warnlaser_logic(Laser *l, int time) {
-	if(time == EVENT_BIRTH) {
-		l->width = 0;
-		return;
-	}
+				if (i > 90) {
+					a *= -1;
+				}
 
-	if(time < 0) {
-		return;
-	}
+				PROJECTILE(
+						.proto = pp_wave,
+						.pos = e->pos,
+						.color = (i & 3) ? RGB(1.0, 0.3, 0.3) : RGB(0.3, 0.3, 1.0),
+						.move = move_linear(2 * dir)
+						);
+				if (global.diff > D_Normal && global.timer % 3 == 0) {
+
+					PROJECTILE(
+							.proto = pp_wave,
+							.pos = e->pos,
+							.color = !(i & 3) ? RGB(1.0, 0.3, 0.3) : RGB(0.3, 0.3, 1.0),
+							.move = move_linear(-2 * dir)
+							);
 
-	if(time == 90) {
-		play_sound_ex("laser1", 30, false);
+				}
+
+				play_sound("shot1");
+				WAIT(shot_interval);
+			}
 	}
 
-	laser_charge(l, time, 90, 10);
-	l->color = *color_lerp(RGBA(0.2, 0.2, 1, 0), RGBA(1, 0.2, 0.2, 0), time / l->deathtime);
+	WAIT(60);
+
+	// fly out
+	cmplx exit_accel = 0.02 * I + (ARGS.side * 0.03);
+	e->move.attraction = 0;
+	e->move.acceleration = exit_accel;
+	e->move.retention = 1;
 }
 
-static void wriggle_ignite_warnlaser(Laser *l) {
-	float f = 6;
-	create_laser(l->pos, 90, 120, RGBA(1, 1, 1, 0), l->prule, wriggle_ignite_warnlaser_logic, f*l->args[0], l->args[1], f*l->args[2], l->args[3]);
+TASK(little_fairy_line, { int count; cmplx pos; cmplx target_pos; }) {
+	for(int i = 0; i < ARGS.count; ++i) {
+		cmplx pos1 = VIEWPORT_W/2+VIEWPORT_W/3*nfrand() + VIEWPORT_H/5*I;
+		cmplx pos2 = VIEWPORT_W/2+50*(i-ARGS.count/2)+VIEWPORT_H/3*I;
+		// type, start pos, end pos, type, side (not needed?)
+		INVOKE_TASK(little_fairy, pos1, pos2, 2);
+		WAIT(5);
+	}
 }
 
-void wriggle_night_ignite(Boss *boss, int time) {
-	TIMER(&time)
+TASK(big_fairy_group, { cmplx pos; int danmaku_type; } ) {
+	// big fairy in the middle does nothing
+	// 8 fairies (2 pairs in 4 waves - bottom/top/bottom/top) spawn around her and open fire
+	Enemy *e = TASK_BIND_UNBOXED(create_enemy1c(ARGS.pos, 10000, BigFairy, NULL, 0));
 
-	float dfactor = global.diff / (float)D_Lunatic;
+	e->alpha = 0;
 
-	if(time == EVENT_DEATH) {
-		enemy_kill_all(&global.enemies);
-		return;
-	}
+	INVOKE_TASK_WHEN(&e->events.killed, common_drop_items, &e->pos, {
+			.points = 3,
+			.power = 2,
+			});
+	WAIT(100);
 
-	if(time < 0) {
-		GO_TO(boss, VIEWPORT_W/2 + VIEWPORT_H*I/3, 0.05)
-		return;
+	for(int x = 0; x < 2; ++x) {
+		// type, big fairy pos, little fairy pos (relative), danmaku type, side of big fairy
```

Comment:
Since the task arguments are actually passed as C structs, you can use the designated initializer syntax instead of writing comments like that. "Keyword arguments" in C, woo.

```c
INVOKE_TASK(little_fairy,
	.pos = e->pos,
	.target_pos = e->pos + 70 + 50 * I,
	.danmaku_type = ARGS.danmaku_type,
	.side = 1
);
```